x-pack/legacy/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_panel.tsx

```diff
@@ -41,152 +50,185 @@ export interface OperationFieldSupportMatrix {
+  // our previous operation)
+  const hasFieldChanged =
+    selectedColumn &&
+    hasField(selectedColumn) &&
+    selectedColumn.sourceField !== droppedItem.field.name &&
+    operationsForNewField &&
+    operationsForNewField.includes(selectedColumn.operationType);
+
+  // If only the field has changed use the onFieldChange method on the operation to get the
+  // new column, otherwise use the regular buildColumn to get a new column.
+  const newColumn = hasFieldChanged
+    ? changeField(selectedColumn, currentIndexPattern, droppedItem.field)
+    : buildColumn({
+        op: operationsForNewField ? operationsForNewField[0] : undefined,
+        columns: props.state.layers[props.layerId].columns,
+        indexPattern: currentIndexPattern,
+        layerId,
+        suggestedPriority: props.suggestedPriority,
+        field: droppedItem.field,
+        previousColumn: selectedColumn,
+      });
+
+  trackUiEvent('drop_onto_dimension');
+  const hasData = Object.values(props.state.layers).some(({ columns }) => columns.length);
+  trackUiEvent(hasData ? 'drop_non_empty' : 'drop_empty');
+
+  props.setState(
+    changeColumn({
+      state: props.state,
+      layerId,
+      columnId: props.columnId,
+      newColumn,
+      // If the field has changed, the onFieldChange method needs to take care of everything including moving
+      // over params. If we create a new column above we want changeColumn to move over params.
+      keepParams: !hasFieldChanged,
+    })
+  );
+
+  return true;
+}
+
+export const IndexPatternDimensionTriggerComponent = function IndexPatternDimensionTrigger(
+  props: IndexPatternDimensionTriggerProps
+) {
+  const layerId = props.layerId;
+
+  const selectedColumn: IndexPatternColumn | null =
+    props.state.layers[layerId].columns[props.columnId] || null;
 
-    return (
-      isDraggedField(dragging) &&
-      layerIndexPatternId === dragging.indexPatternId &&
-      Boolean(hasOperationForField(dragging.field))
-    );
+  const { columnId, uniqueLabel } = props;
+  if (!selectedColumn) {
+    return null;
   }
+  return (
+    <EuiLink
+      id={columnId}
+      className="lnsConfigPanel__triggerLink"
+      onClick={() => {
+        props.togglePopover();
+      }}
+      data-test-subj="lns-dimensionTrigger"
+      aria-label={i18n.translate('xpack.lens.configure.editConfig', {
+        defaultMessage: 'Edit configuration',
+      })}
+      title={i18n.translate('xpack.lens.configure.editConfig', {
+        defaultMessage: 'Edit configuration',
+      })}
+    >
+      {uniqueLabel}
+    </EuiLink>
+  );
+};
+
+export const IndexPatternDimensionEditorComponent = function IndexPatternDimensionPanel(
+  props: IndexPatternDimensionEditorProps
+) {
+  const layerId = props.layerId;
+  const currentIndexPattern =
+    props.state.indexPatterns[props.state.layers[layerId]?.indexPatternId];
+  const operationFieldSupportMatrix = getOperationFieldSupportMatrix(props);
```

Comment:
doesn't it have to be memoized just like it was before? 🤔

Reply:
I would like it to be memoized, but I couldn't find a way to do it safely. Each dimension needs to be memoized separately.